### PR TITLE
Implement MSC3786: Add a default push rule to ignore m.room.server_acl events

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/20788
Implements https://github.com/matrix-org/matrix-spec-proposals/pull/3786

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Review by richvdh — COMMENTED** [review]
I've written some comments on the MSC: I'm not sure why this is needed.

Generally we try to avoid having things using unstable prefixes enabled by default in synapse, so ideally we'd have an `experimental` config option which would enable the additional push rule.

**Comment by SimonBrandner** [comment]
> I've written some comments on the MSC: I'm not sure why this is needed.

I've updated the MSC and replied to some comments

> Generally we try to avoid having things using unstable prefixes enabled by default in synapse, so ideally we'd have an `experimental` config option which would enable the additional push rule.

Since getting the flag requires access to `HomeServerConfig`, I assume I can't just do this in `baserules.py`, would I do this in the constructor of `PushRulesHandler` or is there a better way?

**Comment by SimonBrandner** [comment]
> > > Generally we try to avoid having things using unstable prefixes enabled by default in synapse, so ideally we'd have an `experimental` config option which would enable the additional push rule.
> > 
> > 
> > Since getting the flag requires access to `HomeServerConfig`, I assume I can't just do this in `baserules.py`, would I do this in the constructor of `PushRulesHandler` or is there a better way?
> 
> I think I'd probably filter out the experimental rules at:
> 
> https://github.com/matrix-org/synapse/blob/f90d381c7b93b98a5b6d7353196e2ae0912ea822/synapse/storage/databases/main/push_rule.py#L45-L66
> 
> and heavily comment the new rule in `baserules.py` to say it's behind an experimental config flag?
> 
> I'm not sure if anyone has a better suggestion?

Thank you for the help!

**Comment by SimonBrandner** [comment]
Oops. sorry for the re-request, I didn't realize there was a synapse-core review request too...

File: synapse/storage/databases/main/push_rule.py [hunk 1]
```diff
@@ -54,14 +66,25 @@ def _load_rules(rawrules, enabled_map):
     # We're going to be mutating this a lot, so do a deep copy
     rules = list(list_with_base_rules(ruleslist))
```
[the remark this answers is not in the document]
Comment (SimonBrandner) [reply synapse/storage/databases/main/push_rule.py]: Yeah, that's much nicer - thanks!

**Review by erikjohnston — APPROVED** [review]
This LGTM now, thanks!

**Pull request merged**